compiler.go

```diff
@@ -87,10 +93,18 @@ type compiler struct {
 	codeScratchpad []instruction
 }
 
+func (c *compiler) getScriptOrModule() interface{} {
+	if c.module != nil {
+		return c.module
+	}
+	return c.p // TODO figure comething better
+}
+
```

Comment:
This likely needs something better than returning a simple `Program` object, but I didn't want to make it more complicated than I needed at the time and don't have a good suggestion on what `Script` should contain apart from `Program`.

It also is only needed if you want to provide this as an API as this is basically `getActiveScriptOrModule`